xcube/core/update.py

```diff
@@ -74,8 +75,19 @@ def update_dataset_spatial_attrs(dataset: xr.Dataset,
     :param in_place: If ``True``, *dataset* will be modified in place and returned.
     :return: A new dataset, if *in_place* if ``False`` (default), else the passed and modified *dataset*.
     """
-    return _update_dataset_attrs(dataset, [_LON_ATTRS_DATA, _LAT_ATTRS_DATA],
-                                 update_existing=update_existing, in_place=in_place)
+    if not in_place:
+        dataset = dataset.copy()
+    gm = GridMapping.from_dataset(dataset)
+    is_in_attrs = ['geospatial_lon_min',
+                   'geospatial_lon_max',
+                   'geospatial_lat_min',
+                   'geospatial_lat_max']
+    if update_existing or not all(coord_attr in is_in_attrs for coord_attr in
+                                  dataset.attrs):
```

Comment:
I don't quite understand the purpose of the second test. So if my dataset has attributes other than the four geospatial ones, the existing geospatial attributes will be updated even if I specified `update_existing=False`? If this is the desired behaviour the docstring should explain it.

Reply:
Hi Pont, this is actually code, that Norman wrote and I am not sure.